connectors/sources/generic_database.py

```diff
@@ -36,6 +40,33 @@ def table_filter(table):
     )
 
 
+def validate_db_connection_fields(configuration):
+    """Validate that common database connection parameters are present, non-empty and have the correct types.
+
+    Common database connection parameters:
+        - `host`
+        - `port`
+        - `user`
+        - `password`
+
+    Raises:
+        `ConfigurationFieldMissingError`, if a field is not present
+        `ConfigurationFieldEmptyError`, if a field is empty
+        `ConfigurationFieldTypeError`, if `port` has the wrong type
+    """
+
+    db_server_connection_fields = ["host", "port", "user", "password"]
+
+    validate_non_empty_config_fields(db_server_connection_fields, configuration)
+
+    port = configuration["port"]
+
+    if isinstance(port, str) and not port.isnumeric():
+        raise ConfigurationFieldTypeError(
+            f"Configured port {port} has to be an integer."
+        )
```

Comment:
I feel all these can be moved to `basic_validation`.

Reply:
This is specific to databases, so I don't think it's generic enough to have it in a `basic_validation` method inside `DataSourceConfiguration`.